pkg/virt-handler/options_test.go

```diff
@@ -0,0 +1,343 @@
+						Pages: []*cmdv1.Pages{
+							{Unit: memoryUnit, Size: 4, Count: 314094},
+							{Unit: memoryUnit, Size: 2048, Count: 16},
+							{Unit: memoryUnit, Size: 1048576, Count: 0},
+						},
+						Distances: []*cmdv1.Sibling{
+							{Id: 0, Value: 10},
+							{Id: 1, Value: 10},
+							{Id: 2, Value: 10},
+							{Id: 3, Value: 10},
+						},
+						Cpus: []*cmdv1.CPU{
+							{Id: 0, Siblings: []uint32{0}},
+							{Id: 1, Siblings: []uint32{1}},
+							{Id: 2, Siblings: []uint32{2}},
+							{Id: 3, Siblings: []uint32{3}},
+							{Id: 4, Siblings: []uint32{4}},
+							{Id: 5, Siblings: []uint32{5}},
+						},
+					},
+					{
+						Memory: &cmdv1.Memory{Unit: memoryUnit, Amount: 1223960},
+						Pages: []*cmdv1.Pages{
+							{Unit: memoryUnit, Size: 4, Count: 297798},
+							{Unit: memoryUnit, Size: 2048, Count: 16},
+							{Unit: memoryUnit, Size: 1048576, Count: 0},
+						},
+						Distances: []*cmdv1.Sibling{
+							{Id: 0, Value: 10},
+							{Id: 1, Value: 10},
+							{Id: 2, Value: 10},
+							{Id: 3, Value: 10},
+						},
+						Cpus: []*cmdv1.CPU{
+							{Id: 0, Siblings: []uint32{0}},
+							{Id: 1, Siblings: []uint32{1}},
+							{Id: 2, Siblings: []uint32{2}},
+							{Id: 3, Siblings: []uint32{3}},
+							{Id: 4, Siblings: []uint32{4}},
+							{Id: 5, Siblings: []uint32{5}},
+						},
+					},
+					{
+						Memory: &cmdv1.Memory{Unit: memoryUnit, Amount: 1251752},
+						Pages: []*cmdv1.Pages{
+							{Unit: memoryUnit, Size: 4, Count: 304746},
+							{Unit: memoryUnit, Size: 2048, Count: 16},
+							{Unit: memoryUnit, Size: 1048576, Count: 0},
+						},
+						Distances: []*cmdv1.Sibling{
+							{Id: 0, Value: 10},
+							{Id: 1, Value: 10},
+							{Id: 2, Value: 10},
+							{Id: 3, Value: 10},
+						},
+						Cpus: []*cmdv1.CPU{
+							{Id: 0, Siblings: []uint32{0}},
+							{Id: 1, Siblings: []uint32{1}},
+							{Id: 2, Siblings: []uint32{2}},
+							{Id: 3, Siblings: []uint32{3}},
+							{Id: 4, Siblings: []uint32{4}},
+							{Id: 5, Siblings: []uint32{5}},
+						},
+					},
+					{
+						Memory: &cmdv1.Memory{Unit: memoryUnit, Amount: 1289404},
+						Pages: []*cmdv1.Pages{
+							{Unit: memoryUnit, Size: 4, Count: 314159},
+							{Unit: memoryUnit, Size: 2048, Count: 16},
+							{Unit: memoryUnit, Size: 1048576, Count: 0},
+						},
+						Distances: []*cmdv1.Sibling{
+							{Id: 0, Value: 10},
+							{Id: 1, Value: 10},
+							{Id: 2, Value: 10},
+							{Id: 3, Value: 10},
+						},
+						Cpus: []*cmdv1.CPU{
+							{Id: 0, Siblings: []uint32{0}},
+							{Id: 1, Siblings: []uint32{1}},
+							{Id: 2, Siblings: []uint32{2}},
+							{Id: 3, Siblings: []uint32{3}},
+							{Id: 4, Siblings: []uint32{4}},
+							{Id: 5, Siblings: []uint32{5}},
+						},
+					},
+				},
+			}
+
+			Expect(actualTopology).To(Equal(expectedTopology))
+		})
+		It("should convert api.Capabilities to cmdv1.Topology when certain fields of Capabilities are not initialized", func() {
```

Comment:
@vladikr Could KubeVirt handle partial input?

Reply:
I think it should. Eventually, we are building a set per numa node - in the worst case it will be empty.